receiver/googlecloudmonitoringreceiver/receiver.go

```diff
@@ -285,6 +285,13 @@ func (mr *monitoringReceiver) convertGCPTimeSeriesToMetrics(metrics pmetric.Metr
 			}
 		}
 
+		// Add metric-specific labels if they are present
+		if len(timeSeries.Metric.Labels) > 0 {
+			for k, v := range timeSeries.Metric.Labels {
```

Comment:
Consider using functions to properly handle nil values
```suggestion
			for k, v := range timeSeries.GetMetric().GetLabels() {
```

Reply:
Updated